sickbeard/webserve.py

```diff
@@ -2693,20 +2694,47 @@ def massAddTable(self, rootDir=None):
                             if not indexer_id and i:
                                 indexer_id = i
 
-                cur_dir['existing_info'] = (indexer_id, show_name, indexer)
+                cur_dir['existing_info'] = {"indexer_id": indexer_id or '', "show_name": show_name or '',
+                                            "indexer": indexer or ''}
 
                 if indexer_id and Show.find(sickbeard.showList, indexer_id):
                     cur_dir['added_already'] = True
         return t.render(dirList=dir_list)
 
-    def newShow(self, show_to_add=None, other_shows=None, search_string=None):
+    def newShow(self, show_to_add=None, other_shows=None, search_string=None, **shows_to_add):
         """
         Display the new show page which collects a tvdb id, folder, and extra options and
         posts them to addNewShow
         """
         t = PageTemplate(rh=self, filename="addShows_newShow.mako")
 
-        indexer, show_dir, indexer_id, show_name = self.split_extra_show(show_to_add)
+        if shows_to_add:
+            if isinstance(shows_to_add.items()[0][1], list):
+                shows_from_json = [{} for x in range(len(shows_to_add.items()[0][1]))]
+                for attribute in shows_to_add:
+                    for i, x in enumerate(shows_to_add[attribute]):
+                        try:
+                            shows_from_json[i][attribute] = x
+                        except:
+                            pass
+
+                # Get the first show from the list, and process this one
+                show_to_add = shows_from_json.pop(0)
+                other_shows = shows_from_json
+            else:
+                # This should be the last show to be added
+                show_to_add = shows_to_add
+        elif show_to_add:
+            # Sanitize unquote and eval to dict
+            show_to_add = ast.literal_eval(unquote_plus(show_to_add).strip())
+            other_shows = [ast.literal_eval(unquote_plus(x).strip()) for x in other_shows]
+        else:
+            return
```

Comment:
Need to remove this, or create some sort of error message. But doubt this will ever occur.